### PR TITLE
Playlist

### PR DESCRIPTION
**Pull request by vishalg8454**
This PR adds the functionality of adding videos into playlists. New playlists can also be made.

The components added are -
- `RequireAuth` at src/components/require-auth
- `PlaylistModal` at src/components/modal/playlist-modal
- `PortalForModal` at src/components/portal-for-modal
- `PortalWithPositioning` at src/components/portal-with-positioning
- `PlaylistCard` at src/components/playlist-card

The redux slices added are -
- `playlistSlice.js` at src/store

The pages added are -
- `PlaylistDetailPage` at src/pages/playlist-detail-page
- `PlaylistPage` at src/pages/playlist-page

[live preview](https://deploy-preview-6--videostock.netlify.app/)

Would love to hear your feedback and suggestions.

image

image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -8,8 +8,28 @@ import {
...
+  } = useSelector((state) => state.auth);
+
+  useEffect(() => {
+    if (token) {
+      dispatch(fetchPlaylist(token));
+    }
+  }, [token]);
+
   const view = useMedia(
     ["(min-width: 1000px)", "(min-width: 600px)", "(min-width: 400px)"],
```
Comment (bhavyaj12) [inline src/App.js:34]: Really like this

**Review by bhavyaj12 — APPROVED** [review]
Great work. Code is very well written

**Review by nikhil-sharma-nks — APPROVED** [review]
Your code structure and code flow is predictable and easy to understand. Good work with UI. Just need to remove commented code that you have and everything looks good to be merged.

File: src/App.js (hunk 1 above)
Comment (vishalg8454) [reply src/App.js:34]: Thank you

**Pull request merged**